packages/gradle-plugin/src/main/kotlin/io/realm/kotlin/gradle/analytics/AnalyticsService.kt

```diff
@@ -39,63 +50,330 @@ import org.gradle.tooling.events.task.TaskSuccessResult
  *
  * **See:** [Build Services](https://docs.gradle.org/current/userguide/build_services.html)
  */
-abstract class AnalyticsService : BuildService<BuildServiceParameters.None>, OperationCompletionListener {
 
-    private val logger: Logger = Logging.getLogger("realm-build-service")
-    private var analytics: RealmAnalytics? = null
+// Asynchronously submits build information to Realm when the gradle compile task run
+//
+// To be clear: this does *not* run when your app is in production or on
+// your end-user's devices; it will only run when you build your app from source.
+//
+// Why are we doing this? Because it helps us build a better product for you.
+// None of the data personally identifies you, your employer or your app, but it
+// *will* help us understand what Realm version you use, what host OS you use,
+// etc. Having this info will help with prioritizing our time, adding new
+// features and deprecating old features. Collecting an anonymized bundle &
+// anonymized MAC is the only way for us to count actual usage of the other
+// metrics accurately. If we don't have a way to deduplicate the info reported,
+// it will be useless, as a single developer building their app on Windows ten
+// times would report 10 times more than a single developer that only builds
+// once from Mac OS X, making the data all but useless. No one likes sharing
+// data unless it's necessary, we get it, and we've debated adding this for a
+// long long time. Since Realm is a free product without an email signup, we
+// feel this is a necessary step so we can collect relevant data to build a
+// better product for you.
+//
+// Currently the following information is reported:
+// - What version of Realm is being used
+// - What host you are running on
+// - What targets you are building for
+// - An anonymized MAC address and bundle ID to aggregate the other information on.
+//
+// The collected information can be inspected by settings the system environment variable
+//   REALM_PRINT_ANALYTICS=true
```

Comment:
Nitpick. It feels a bit weird to control output this way. I would assume this was just controlled by either enabling debug or verbose log level

Reply:
It is part of the scope to do it this way, so maybe that was why I printed it to the console (warning-level) 🤔 We could also just dump it unconditionally on info? 
It seems extreme to require people to enable debug log-level to inspect this, as they will be flooded with all other debug information.